File: frontend/src/types/subscription.ts
export type BillingCycle = 'MONTHLY' | 'YEARLY' | 'WEEKLY'

export const BILLING_CYCLE_LABELS: Record<BillingCycle, string> = {
  MONTHLY: '월간',
  YEARLY: '연간',
  WEEKLY: '주간',
}

export interface Subscription {
  id: number
  name: string
  startDate: string
  billingCycle: BillingCycle
  amount: number
  nextBillingDayOfMonth: number | null
  createdAt: string
}

export interface SubscriptionRequest {
  name: string
  startDate: string
  billingCycle: BillingCycle
  amount: number
}

File: backend/src/main/java/com/cashbook/domain/subscription/entity/Subscription.java
package com.cashbook.domain.subscription.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.OffsetDateTime;

@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Subscription {
    private Long id;
    private Long userId;
    private String name;
    private LocalDate startDate;
    private String billingCycle;  // MONTHLY, YEARLY, WEEKLY
    private Long amount;
    private OffsetDateTime createdAt;
    private OffsetDateTime updatedAt;
    private OffsetDateTime deletedAt;
}
